app/src/shared/hooks/usePreventNavigatingAway.ts

```diff
@@ -85,24 +92,64 @@ export default function usePreventNavigatingAway({
     )
 }
 
+function patchBeforeUnload() {
+    if (window.addEventListener !== Window.prototype.addEventListener) {
+        return
+    }
+
+    function addEventListener(type: any, ...rest: [any, any]) {
+        if (type === 'beforeunload') {
+            return
+        }
+
+        Window.prototype.addEventListener.call(window, type.replace(/^hub_/, ''), ...rest)
+    }
+
+    window.addEventListener = addEventListener
+
+    function removeEventListener(type: any, ...rest: [any, any]) {
+        if (type === 'beforeunload') {
+            return
+        }
+
+        Window.prototype.removeEventListener.call(
+            window,
+            type.replace(/^hub_/, ''),
+            ...rest,
+        )
+    }
+
+    window.removeEventListener = removeEventListener
+}
+
```

Comment:
It's been a while since I had to do so much hackery.

The major breaking thing in the router library (or rather the `history` library) is… blocking navigation.

`history.block` does few stupid things:
- Forces its own `beforeunload` event handler on you and it's... raw, unconditioned, unreasonable. In other words, it always asks to confirm.
- Discards blockers after a single use, which is... why would you do that eagerly I have no idea!

v6 is not as elegant as I'd like. Hope it's just a phase.

Here I overwrite `window.(add|remove)EventListener` to ignore `beforeunload` events, and I introduce `hub_beforeunload` so that only our custom events and handlers matter.

I'mma keep an eye on the library or implement the blocking piece properly myself if I find the time.